compiler-plugin-tests/src/test/resources/ballerina_sources/valid_service_11/service.bal

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/graphql;
+
+// NamedEntity and SomeNamedEntity are not being used in the graphql service, not validated as graphql interfaces
+service class NamedEntity {
+    string name;
+
+    function init(string name) {
+        self.name = name;
+    }
+
+    isolated resource function get name() returns string {
+        return self.name;
+    }
+}
+
+service class SomeNamedEntity {
+    string name;
+
+    function init(string name) {
+        self.name = name;
+    }
+
+    isolated resource function get name() returns string {
+        return self.name;
+    }
+}
+
+distinct service class ContactDetail {
+    string email;
+
+    function init(string email) {
+        self.email = email;
+    }
+
+    isolated resource function get email() returns string {
+        return self.email;
+    }
+}
+
+distinct service class Person {
```

Comment:
This seems wrong. Consider we have a distinct service class used as a GraphQL interface. Then it must have resource functions, otherwise there's no point of defining it as an interface, right?
In this case, defining a `Person` class is redundant. They can directly use the `ContactDetail` class. I think we should not allow this.

Reply:
Fixed in 7083720d8f6cbcb64c2120100f26aaf3abfa5fe4

Reply:
Fixed in 7083720d8f6cbcb64c2120100f26aaf3abfa5fe4